src/components/register/ServerModal.tsx

```diff
@@ -16,6 +27,63 @@ const ServerModal = () => {
     setIsOpen(true);
   };
 
+  const saveModal = () => {
+    refetch();
+  };
+
+  useEffect(() => {
+    if (data && !isStale && isOpen) {
+      settings.dispatch({
+        type: 'SET_SERVER',
+        payload: { server: serverAddress },
+      });
+
+      setTimeout(() => {
+        closeModal();
+      }, 500);
```

Comment:
L'utilisation de useEffect me parait un peu overkill voir dangereuse dans cette situation. 
`If you’re familiar with React class lifecycle methods, you can think of useEffect Hook as componentDidMount, componentDidUpdate, and componentWillUnmount combined.`

Du coup elle proc à chaque fois que tu tapes dans l'input ?

Et même, ça veut dire que ton bouton ne proc pas directement le changement de serveur mais c'est plus un side effect d'un `componentDidUpdate` du coup pour la compréhension du code c'est pas clair.

Il faudrait juste une fonction qui proc sur le onClick du bouton pour dispatch la nouvelle adresse, refetch et close la modal. Au moins ce serait plus clair et tu es súr de pas avoir de side effect à la con.

Après c'est peu être juste parce que je suis plus habitué à de l'Angular et sur des projets à plus grosse échelle.

Reply:
Oui useEffect fait en théorie tout ça mais c'est en fonction des arguments, avec [], c'est uniquement `componentDidMount`, avec [var1, var2], c'est `componentDidMount` et `componentDidUpdate` uniquement si les dépendences changent et `return () => { some cleanup }` c'est `componentWillUnmount`. Gérer une réponse asynchrone dans le bouton block l'ui et je vois pas de façon propre de le faire. Le bouton fait la requête et l'ui s'adapte a l'état de la requête. Le useEffect change le server dans les settings une fois qu'il y a des données, qu'elles sont récentes et que le modal est ouvert, après 500ms, une fois que tout est fait, le modal ce ferme

Reply:
et il se passe quoi si pour une raison random le serveur met plus de 500ms à répondre ?

Reply:
Rien